python/dgl/data/citation_graph.py

```diff
@@ -118,8 +118,13 @@ def process(self):
 
         if self.reverse_edge:
             graph = nx.DiGraph(nx.from_dict_of_lists(graph))
+            g = from_networkx(graph)
         else:
             graph = nx.Graph(nx.from_dict_of_lists(graph))
+            edges = list(graph.edges())
+            u, v = map(list, zip(*edges))
+            g = dgl_graph((u, v))
+            graph = g.to_networkx()
```

Comment:
Even if the fix works, the change is very convoluted. You are creating a networkx graph, converting it to DGLGraph and then converting it back to networkx graph. Why can't we just have networkx graph -> DGLGraph? Please simplify the code.

Reply:
1. This class used to have an attribute `._graph` that is a NetworkX graph, do we still want it?
2. The NetworkX graph at L123 is an undirected one and the implementation obtained edges for one direction by taking `graph.edges()`. However, this will not give a directed NetworkX graph with edges for a single direction, hence L127.

Reply:
For 2, the original implementation relies on `NetworkX` to drop half of the edges as the graph loaded by pickle has edges for both directions.

Reply:
Anyway, the goal is to have simpler implementation. You could start with a networkx graph to perform some pruning or processing and convert to DGLGraph later. Or you could start with DGLGraph directly and use DGL APIs to achieve the same logics. Either way, just do not have networkx->DGLGraph->networkx->DGLGraph multiple times.

Reply:
I've removed the `._graph` attribute as it does not seem to be used anywhere. This avoids reconverting the `DGLGraph` into a NetworkX graph in the end. @jermainewang